### PR TITLE
Random.shuffle(): mattn suggestion implementation

### PR DESCRIPTION
**Pull request by tsuyoshicho**
see https://vim-jp.slack.com/archives/C8F0X4DHN/p1562239287236300

This implementation is technical trial.

Please review it.

### CODE REVIEW

**Comment by tsuyoshicho** [comment]
#677 happen.

File: autoload/vital/__vital__/Random.vim [hunk 1]
```diff
@@ -70,16 +70,11 @@ function! s:Random.shuffle(list) abort
   while 1 < pos
     let n = self.range(pos)
     let pos -= 1
-    if n != pos
-      let temp = a:list[n]
-      let a:list[n] = a:list[pos]
-      let a:list[pos] = temp
-    endif
+    let [a:list[n], a:list[pos]] = [a:list[pos], a:list[n]]
```
Comment (thinca) [inline autoload/vital/__vital__/Random.vim]: Using list assignment to swap is OK. 👌
I have doubts about removing `if n != pos`. 🤔
There is needless swapping when it is removed.

Comment (tsuyoshicho) [reply autoload/vital/__vital__/Random.vim]: Hmmm.

Okay, re-fix for later.

Comment (tsuyoshicho) [reply autoload/vital/__vital__/Random.vim]: fixit

**Pull request merged**